plugin/cache.go

```diff
@@ -221,9 +221,13 @@ func cachedOp(opName string, entry Entry, ttl time.Duration, op opFunc) (interfa
 	}
 
 	if entry.id() == "" {
-		// The Registry's ID is set to "/", while all other entries' IDs are
-		// set in CachedList. Thus, we should never hit this code-path.
-		panic("entry.id() returned an empty ID")
+		// This can happen if the entry's just been created, and it calls one of the Cached*
+		// methods to query its Metadata so that it can set its initial attributes. In that case,
+		// just invoke op. Ideally we should panic here, but panicking makes it significantly more
+		// annoying for plugin authors to write their code. Since the plugin package manages the
+		// entry's ID, and the entry's ID is set in CachedList, an empty ID is such a rare edge case
+		// that it's better to not annoy people.
+		return op()
```

Comment:
The panic behavior led me to either moving the cached op to the parent, or creating separate uncached/cached methods, where I'd use the former to compute the initial attributes in a parent's List call, and the latter for subsequent operations.

Reply:
Ah right, because they're using their own caching now. This seems like a reasonable compromise.